app/com/arpnetworking/metrics/portal/hosts/impl/ForemanClient.java

```diff
@@ -0,0 +1,290 @@
+        }
+
+        public Integer getSubtotal() {
+            return _subtotal;
+        }
+
+        public Integer getPage() {
+            return _page;
+        }
+
+        public Integer getPerPage() {
+            return _perPage;
+        }
+
+        public List<ForemanHost> getResults() {
+            return _results;
+        }
+
+        private HostPageResponse(final Builder builder) {
+            _total = builder._total;
+            _subtotal = builder._subtotal;
+            _page = builder._page;
+            _perPage = builder._perPage;
+            _results = builder._results;
+        }
+
+        private final Integer _total;
+        private final Integer _subtotal;
+        private final Integer _page;
+        private final Integer _perPage;
+        private final List<ForemanHost> _results;
+
+        /**
+         * Implementation of the builder pattern for {@link HostPageResponse}.
+         *
+         * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+         */
+        public static final class Builder extends OvalBuilder<HostPageResponse> {
+            /**
+             * Public constructor.
+             */
+            public Builder() {
+                super(HostPageResponse::new);
+            }
+
+            /**
+             * Sets the total number of hosts.  Required. Cannot be null.
+             *
+             * @param value The total number of hosts
+             * @return this {@link Builder}
+             */
+            public Builder setTotal(final Integer value) {
+                _total = value;
+                return this;
+            }
+
+            /**
+             * Sets the Subtotal.  Required. Cannot be null.
+             *
+             * @param value The subtotal
+             * @return this {@link Builder}
+             */
+            public Builder setSubtotal(final Integer value) {
+                _subtotal = value;
+                return this;
+            }
+
+            /**
+             * Sets the current page.  Required. Cannot be null.
+             *
+             * @param value The current page number (1-based)
+             * @return this {@link Builder}
+             */
+            public Builder setPage(final Integer value) {
+                _page = value;
+                return this;
+            }
+
+            /**
+             * Sets the number of hosts per page.  Required. Cannot be null.
+             *
+             * @param value Number of hosts per page
+             * @return this {@link Builder}
+             */
+            public Builder setPerPage(final Integer value) {
+                _perPage = value;
+                return this;
+            }
+
+            /**
+             * Sets the list of hosts.  Required. Cannot be null.
+             *
+             * @param value List of hosts
+             * @return this {@link Builder}
+             */
+            public Builder setResults(final List<ForemanHost> value) {
```

Comment:
I've been moving towards Guava's ImmutableList and ImmutableSet more with much nicer semantics wrt copies. I would suggest either that or then a copy in the pojo constructor above.